api/devicerepository.proto

```diff
@@ -0,0 +1,388 @@
+    map<string, Profile> profiles = 4;
+  }
+  // Available firmware versions.
+  repeated FirmwareVersion firmware_versions = 6;
+  // List of sensors included in the device.
+  repeated string sensors = 7;
+  message Dimensions {
+    // Device width (mm).
+    google.protobuf.FloatValue width = 1;
+    // Device height (mm).
+    google.protobuf.FloatValue height = 2;
+    // Device diameter (mm).
+    google.protobuf.FloatValue diameter = 3;
+    // Device length (mm).
+    google.protobuf.FloatValue length = 4;
+  }
+  // Device dimensions.
+  Dimensions dimensions = 8;
+  // Device weight (gram).
+  google.protobuf.FloatValue weight = 9;
+  message Battery {
+    // Whether the device battery can be replaced.
+    google.protobuf.BoolValue replaceable = 1;
+    // Battery type.
+    string type = 2;
+  }
+  // Device battery information.
+  Battery battery = 10;
+  message OperatingConditions {
+    message Limits {
+      // Min value of operating conditions range.
+      google.protobuf.FloatValue min = 1;
+      // Max value of operating conditions range.
+      google.protobuf.FloatValue max = 2;
+    }
+    // Temperature operating conditions.
+    Limits temperature = 1;
+    // Relative humidity operating conditions.
+    Limits relative_humidity = 2;
+  }
+  // Device operating conditions.
+  OperatingConditions operating_conditions = 11;
+  // Device IP rating code.
+  string ip_code = 12 [(gogoproto.customname) = "IPCode"];
+  // Supported key provisioning methods.
+  repeated KeyProvisioning key_provisioning = 13;
+  // Device key security.
+  KeySecurity key_security = 14;
+
+  message Photos {
+    // Main device photo.
+    string main = 1;
+    // List of other device photos.
+    repeated string other = 2;
+  }
+  // Device photos.
+  Photos photos = 15;
+  message Videos {
+    // Link to main device video.
+    string main = 1;
+    // Links to other device videos.
+    repeated string other = 2;
+  }
+  // Device videos.
+  Videos videos = 16;
+
+  // Device information page URL.
+  string product_url = 17 [(gogoproto.customname) = "ProductURL"];
+  // Device datasheet URL.
+  string datasheet_url = 18 [(gogoproto.customname) = "DatasheetURL"];
+
+  message Compliances {
+    message Compliance {
+      string body = 1;
+      string norm = 2;
+      string standard = 3;
+      string version = 4;
+    }
+    // List of safety standards the device is compliant with.
+    repeated Compliance safety = 1;
+    // List of radio equipment standards the device is compliant with.
+    repeated Compliance radio_equipment = 2;
+  }
+  // List of standards the device is compliant with.
+  Compliances compliances = 19;
+
+  // List of any additional radios included in the device.
+  repeated string additional_radios = 20;
+}
+
+message GetEndDeviceBrandRequest {
+  // Application identifiers.
+  ApplicationIdentifiers application_ids = 1 [(gogoproto.customname) = "ApplicationIDs"];
```

Comment:
What are these for? Also in other get requests

Reply:
This is for authentication purposes. We cannot require application rights without application identifiers.

Reply:
Right, but why make it open?

Reply:
What do you mean by "make it open"?

Reply:
See below